pkg/extra/service.go

```diff
@@ -2,45 +2,24 @@ package extra
 
 import (
 	"context"
-	"net/http"
+
+	hClient "github.com/ory/hydra-client-go/v2"
+	kClient "github.com/ory/kratos-client-go"
 
 	"github.com/canonical/identity-platform-login-ui/internal/logging"
 	misc "github.com/canonical/identity-platform-login-ui/internal/misc/http"
 	"github.com/canonical/identity-platform-login-ui/internal/monitoring"
 	"github.com/canonical/identity-platform-login-ui/internal/tracing"
-
-	hClient "github.com/ory/hydra-client-go/v2"
-	kClient "github.com/ory/kratos-client-go"
 )
 
 type Service struct {
-	kratos KratosClientInterface
-	hydra  HydraClientInterface
+	hydra HydraClientInterface
 
 	tracer  tracing.TracingInterface
 	monitor monitoring.MonitorInterface
 	logger  logging.LoggerInterface
 }
 
-func (s *Service) CheckSession(ctx context.Context, cookies []*http.Cookie) (*kClient.Session, error) {
-	ctx, span := s.tracer.Start(ctx, "kratos.FrontendApi.ToSession")
-	defer span.End()
-
-	session, r, err := s.kratos.FrontendApi().ToSession(
-		ctx,
-	).Cookie(
-		misc.CookiesToString(cookies),
-	).Execute()
-
-	if err != nil {
-		// TODO @shipperizer we shouldn't be logging this
-		s.logger.Debugf("full HTTP response: %v", r)
-
-		return nil, err
-	}
-	return session, nil
-}
```

Comment:
So I removed this because it was copy/paste code, logic was repeated without any difference. Also and more importantly hydra service was relying on both hydra client and kratos client, which is against the single responsibility principle.

Having the controller layer rely on only one service is not a "practice", although in our codebase it appears like that. That would bring a bloated single service that has to do everything the controller (handler) needs.

It is perfectly normal/common to have a handler/controller orchestrate responses from multiple services, with each service doing "their own" singly responsible thing.

What you don't want is same-layer services being mutually dependant on one another.